### ISSUE TITLE
Not rendering / Executing properly and also High cpu usage

### ISSUE THREAD

**Issue by ZeusThundr** [karlstav/cava#446]
**When/where was the bug introduced?**
Unsure

**Describe the bug**
High cpu usage wont load and or execute. Only executes if built from source I have followed the build guide to the letter for version 0.8 the void repos use version cava-0.7.5_1. Just looked through some other issues and just double checked to make sure that my locale was set properly and I to described this better 0.8 just leaves me on a blank screen and version 0.7.5_1 is in the screenshot.

`LC_ALL=en_US.UTF-8 ` 
is set in /etc/locale.conf as well as some other options

**To Reproduce**
just execute the program

**My Current Testing Config**
I have used alsa_zero_test, fifo_zero_test, pulse_zero_test, my own config from desktop setup (arch), and the default config.
```
[general]
framerate = 1
autosens = 0
sensitivity = 100
bars = 4
; bars = 2

[input]
method = fifo
source = /tmp/mpd.fifo
; source = /dev/zero

[output]
method = ncurses
; method = noncurses
; method = raw

[color]
gradient = 0
```
**Other things I have tried**
0.8 built from sources executes but just leave a blank screen unless I try to launch with method that it unsupported like sdl
I have built it many times from sources with different optional components non work that I have found and when I press CTRL-C it always says goodbye on either 0.7.5_1 or 0.8

**MPD**
I have changed the mpd config
```
audio_output {
               type "fifo"
               name "Visualizer feed"
               path "/tmp/mpd.fifo"
               format "22050:16:2"
}
```
the format I have tried 44100 and 48000

**Expected behavior**
to render and or execute

**Screenshots**
![cava](https://user-images.githubusercontent.com/76399616/163654623-7ddffba7-9b22-437e-ad73-f41dc09a61dd.png)

**Desktop (please complete the following information):**
Void Linux - rpi4 (aarch64) 2021-09-30
Live image musl

**Terminal emulators tried**
Tried: XTerm, URxvt, Kitty, and Alacritty

**Audio setups tried**
Pulseaudio (stand alone)
Pipewire w/pipewire media session ( setup as pulseaudio replacement)
Pipewire w/wireplumber ( setup as pulseaudio replacement)
Alsa (without loopback - not configured properly for cava)
I can confirm pipewire is setup properly as pulse replacement as `pactl info` indicates as well as pulsemixer works fine and 
`pamixer -i 5` works perfectly as well.

I have used ncmpcpp, glava, mpdviz all work as intended I have also tried some other audio sources besides just mpd ( mpv, firefox )

**Current setup**
URxvt-unicode-9.30_2
Pipewire-0.3.48_1
Wireplumber-0.4.9_1
Default shell: Dash-0.5.11.5_1
Interactive: Fish-shell-3.4.1_1

**Additional context and some side notes**
I just wrote all the configs in this markdown file by hand so if you note a mistake/typo I promise you it is on written incorrently on the pi because I didn't copy and paste the configs as I am on another machine right now. -- excuses im just lazy :P
I might of missed something had to write this twice as I acciently reloaded the page the first time I wrote this.
Its probably a pain to implement as I am not a programmer but multithreaded wout be nice to see <3
The raspberry pi is overclocked to 2.3 ghz glava is using mesa driver and runs about 42 fps ( just adding this line to help people search for bug reports on the rpi) I saw someone talk about adding opengl as a dependency for installing for the pi but I can assure you that the driver is working fine on my pi as alacritty, kitty and glava work fine. ( I forget sure if  they said mesa or opengl)

**Comment by karlstav** [comment]
the test configs are only meant for internal testing.

can you please start with the example config [file](https://github.com/karlstav/cava/blob/master/example_files/config) set the desired input method (looks like it's mpd) and source, nothing more!

use 44100 rate in mpd config.

**Comment by ZeusThundr** [comment]
I have changed mpd config to 44100 and I am not using the example config I have tried pulse method and fifo the only thing of note is that now my terminal's cursor is blinking when I launch cava where as before it was not.

**Comment by karlstav** [comment]
what is your output of `localectl` command?

**Comment by ZeusThundr** [comment]
here is my locale.conf I think localectl might be from systemd don't have that command
```
LC_ALL=en_US.UTF-8
LANG=en_US.UTF-8
LC_COLLATE=C
```

**Comment by karlstav** [comment]
to just test that the locale stuff is ok can you run

```
echo -e '\u2586'
```

in your terminal?

it should output a 6/8 block like this: `▆`.

**Comment by ZeusThundr** [comment]
Interesting I just got back \u2586

**Comment by karlstav** [comment]
> and when running cava inside of bash or zsh it exhibits the same behaviour

so cava works fine in bash or zsh?

**Comment by ZeusThundr** [comment]
No the same beheviour as previously mentioned high cpu usage and not executing/workin. Side not I honestly think It might be an Xorg driver problem on the Void linux for the pi. I might try it to install rasbian and then maybe manjaro and get back to you. as I have tried to get the _xf86-video-fbturbo_ driver working (Xorg DDX driver for arm devices) to no luck the driver seems to be unmaintained commit being 7 years ago. I am not sure how that driver is supposed to help as mostly everything seems to work fine but a **HINT** might lie in the _libEGL_ warnings I recieve when launching certain opengl applications stating "failed to authenticate" not sure what any of that really means as I am not a developer or programmer I just use linux for fun so take what   I am saying with a grain of salt. I have also noted that when I am using full kms "an experimental GL driver" for the pi alacrrity and kitty will not launch. With my current setup I know that with my current setup _SDL2_ and _opengl_ work due to the fact the pi can run binaries like minetest, alacritty, and chocolate-doom, and glava(distort audio through headphone jack not usb not sure why and slows down the system signicantly), hence the reason I would like to use cava for the gradient to match xresources and it being more light weight to run + the rice factor).
**TLDR: Just close the issue I do not think it is your fault**

**Comment by karlstav** [comment]
well since this issue popped up after i made a lot of big changes it was a bit disturbing. if sdl and opengl works, then the sdl output of cava should work. but i do suspect a locale issue keeping the ncruses output from working.

if you like you could check out the latest 0.7 release and see if it works

```
git checkout 0.7.5
make clean
sudo make uninstall
make
sudo make install
```

**Comment by ZeusThundr** [comment]
the version in the screenshot is 0.7.5 from source and it is also the same version the official void repos neither one works same behavior as before just compiled version 0.8.2-1 and it also is exhibiting the same issues as noted before. sorry for the late reply and thank you trying to help.
